### ISSUE TITLE
Custom ConfigureMemberCookieOptions doesn't work anymore 

### ISSUE THREAD

**Issue by JeroenRoos** [umbraco/Umbraco-CMS#14377]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.4.0

### Bug summary

This is while using the following attribute: `[UmbracoMemberAuthorize]`

When on version 11.2.2, I added a `ConfigureCustomMemberCookieOptions` class which redirects the Umbraco Member to the login page, when the Member wasn't authorized to view a specific page. This class is registered in a Composer.

After I upgraded my local application to version 11.4.0, that same class didn't work anymore, and instead of a redirect to the login page, I get a page that just shows the following: HTTP ERROR 401. 

I didn't change anything to the implementation, and I still have a working version of this running on a staging environment, where the Member is redirected to the login environment.

I looked at the changes, and maybe it might have something to do with the following bug fix?
https://github.com/umbraco/Umbraco-CMS/pull/14036/files/339c9c89cb8f9f17e257719632f460cdbf8fdeda

### Specifics

**A simplified version of my custom ConfigureMemberCookieOptions implementation**

![Custom ConfigureMemberCookieOptions Implementation](https://github.com/umbraco/Umbraco-CMS/assets/25620488/575bc6ae-5c43-4ff4-9203-9743ac9d6c18)

**Registering in the Composer**
`builder.Services.ConfigureOptions<ConfigureCustomMemberCookieOptions>();`

### Steps to reproduce

Add a class which implements `IConfigureNamedOptions<CookieAuthenticationOptions>` and configure this in startup/composer (see Specifics). 

In Umbraco 11.2.2, this will make sure that the Member is redirected to a specific page, if the Member isn't authorized.

In Umbraco 11.4.0 this will render a basic 401 error page with the message:
This page isn’t working
If the problem continues, contact the site owner.
HTTP ERROR 401 

---
_This item has been added to our backlog [AB#30478](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/30478)_

**Comment by Zeegaan** [comment]
How are you utilizing the `[UmbracoMemberAuthorize]` attribute? Do you have a sample controller i could use here ? 😁

**Comment by JeroenRoos** [comment]
I have a RenderController like this:

```
[UmbracoMemberAuthorize]
    public abstract class AuthorizedRenderController : RenderController
    {

        protected AuthorizedRenderController(ILogger<RenderController> logger,
            ICompositeViewEngine compositeViewEngine,
            IUmbracoContextAccessor umbracoContextAccessor)
            : base(logger, compositeViewEngine, umbracoContextAccessor)
        {
        }
    }
```

Which is used by other RenderControllers:
`public class BaseReportController : AuthorizedRenderController`

**Comment by Zeegaan** [comment]
Thanks for the sample code, I can definitely reproduce this!

**Comment by mwillebrands** [comment]
@Zeegaan, I think I fixed this in #14399

**Comment by Zeegaan** [comment]
@mwillebrands Sounds great! I will leave a comment on there to make sure we get this tested 💪

**Comment by elit0451** [comment]
Fixed in: #14399
